File: engines/java/src/main/java/io/github/davecookvectorlabs/windowcleaning/WindowType.java
package io.github.davecookvectorlabs.windowcleaning;

/**
 * Window types and their base cleaning time in minutes per window.
 * Values reflect typical commercial window cleaning durations for
 * Northern Ontario commercial facilities.
 */
public enum WindowType {
    STANDARD_SINGLE_PANE(4.0),
    DOUBLE_PANE(5.0),
    FLOOR_TO_CEILING(8.0),
    STOREFRONT(6.0),
    SPECIALTY(12.0);

    private final double baseMinutes;

    WindowType(double baseMinutes) {
        this.baseMinutes = baseMinutes;
    }

    public double getBaseMinutes() {
        return baseMinutes;
    }
}

File: engines/php/src/WindowType.php
<?php

declare(strict_types=1);

namespace WindowCleaning;

/**
 * Window types and their base cleaning time in minutes.
 */
enum WindowType: string
{
    case StandardSinglePane = 'standard_single_pane';
    case DoublePane = 'double_pane';
    case FloorToCeiling = 'floor_to_ceiling';
    case Storefront = 'storefront';
    case Specialty = 'specialty';

    public function baseMinutes(): float
    {
        return match ($this) {
            self::StandardSinglePane => 4.0,
            self::DoublePane => 5.0,
            self::FloorToCeiling => 8.0,
            self::Storefront => 6.0,
            self::Specialty => 12.0,
        };
    }
}
